### PR TITLE
improve testing automation

### LINKED ISSUE #9: Improve testing automation

### PR DESCRIPTION
**Pull request by justinprieto**
Improve testing automation using tox and github actions
Add formatting and linting tox setup

Closes #9

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
+on: [push, pull_request]
 
 jobs:
-  all:
-
+  tests:
     runs-on: ubuntu-latest
```
Comment (pjones-cedar) [inline .github/workflows/tests.yml:6]: Is there a lighter image you can run this on? Or does it not make a difference

File: test.sh [hunk 2]
```diff
@@ -8,24 +11,13 @@ print_in_cyan() {
     printf "\x1B[36m%b\x1B[0m\n" "$*"
 }
 
-PYTHON_VERSIONS="3.8  3.9"
-TEST_SQLALCHEMY=true
-TEST_DJANGO=true
+PYTHON_VERSIONS=('3.8' '3.9')
```
Comment (pjones-cedar) [inline test.sh:14]: Suggest testing on 3.7 and 3.10 aka the oldest supported and also latest release. If something breaks you can narrow it down from there. Skip 3.8 and 3.9 because you can induce that if it passes on both 3.7 and 3.10 it will pass on those as well

**Review by pjones-cedar — APPROVED** [review]
- [x] I fully understand what the code is doing and its context
- [x] The code is the best way to accomplish this problem (I would code it similarly)
- [x] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [x] The code handles edge cases (that I can think of) and has tests to validate them
- [x] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)

File: .github/workflows/tests.yml (hunk 1 above)
Comment (justinprieto) [reply .github/workflows/tests.yml:6]: Shouldn't make a huge diff since it's already pretty fast. If anything this is faster because it's used a lot so might be cached.

Comment (pjones-cedar) [reply .github/workflows/tests.yml:6]: makes sense

File: test.sh (hunk 2 above)
Comment (justinprieto) [reply test.sh:14]: 3.10 is coming in a future release. There are some issues with 3.6 and 3.7 because internal python changes changed how the stacktraces are computed, so the snapshots break. We're thinking about a way to solve it.

**Review by juanrgon — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -1,19 +1,24 @@
 name: Tests
-
-on:
-  push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
+on: [push, pull_request]
```
Comment (juanrgon) [inline .github/workflows/tests.yml:2]: Not important, but this isn't required on every push to every branch
